confd/service/service.go

```diff
@@ -277,69 +88,57 @@ func isModificationAction(action string) bool {
 	return confd.ContainsString(modificationActions, action)
 }
 
-func isServiceNode(node *client.Node, tag string) (bool, error) {
-	if node == nil {
-		return false, nil
-	}
-
-	if node.Value == "" {
-		return false, nil
-	}
-
-	service, err := convertToService(tag, node.Value)
-	if err != nil {
-		return false, errors.Wrap(err, "failed to convert node to service")
-	}
-
-	return service != nil, nil
-}
-
-func reloadServicesIfNecessary(resp *client.Response, conf Configuration, registry configRegistry.Registry) {
+func reloadServicesIfNecessary(loader *Loader, resp *client.Response) {
 	key := resp.Node.Key
-	changed, err := hasServiceChanged(conf, resp)
+	changed, err := loader.HasServiceChanged(resp)
 	if err != nil {
 		log.Printf("failed to check if the change is responsible for a service: %v", err)
-		reloadServices(conf, registry)
+		loader.ReloadServices()
 		return
 	}
 
 	action := resp.Action
 
 	if changed {
 		log.Printf("service %s changed, action=%s", key, action)
-		reloadServices(conf, registry)
+		loader.ReloadServices()
 	} else {
 		log.Printf("ignoring change to non service key %s with action %s", key, action)
 	}
 }
 
 // Run creates the configuration for the services and updates the configuration whenever a service changed
 func Run(conf Configuration, registry configRegistry.Registry) {
-
 	serviceChannel := make(chan *client.Response)
 	maintenanceChannel := make(chan *client.Response)
+	loader := &Loader{
+		registry: registry,
+		config:   conf,
+		writer:   &CommandWriter{config: conf},
+	}
 
 	log.Println("starting service watcher")
 	go func() {
 		for {
-			reloadServices(conf, registry)
+			loader.ReloadServices()
 			registry.Watch(conf.Source.Path, true, serviceChannel)
 		}
 	}()
 	log.Println("starting maintenance mode watcher")
 
 	go func() {
 		for {
-			reloadServices(conf, registry)
+			// TODO: necessary?
+			loader.ReloadServices()
```

Comment:
The ReloadServices is already called at the top goroutines. IDK, maybe reload the services and then start both watchers?